### PR TITLE
Add test for 'COMMENT ON COLUMN' statement

### PR DESCRIPTION
**Pull request by findinpath**
Test infrastructure PR.

### Related pull requests

Extracted from #13800

### CODE REVIEW

File: core/trino-main/src/test/java/io/trino/execution/BaseDataDefinitionTaskTest.java [hunk 1]
```diff
@@ -139,6 +141,11 @@ protected static QualifiedName qualifiedName(String name)
         return QualifiedName.of(TEST_CATALOG_NAME, SCHEMA, name);
     }
 
+    protected static QualifiedName qualifiedColumnName(String objectName, String columnName)
```
Comment (ebyhr) [inline core/trino-main/src/test/java/io/trino/execution/BaseDataDefinitionTaskTest.java:144]: nit: Let's rename `objectName` to `tableName`.

File: core/trino-main/src/test/java/io/trino/execution/BaseDataDefinitionTaskTest.java [hunk 2]
```diff
@@ -420,6 +427,21 @@ public void setViewComment(Session session, QualifiedObjectName viewName, Option
...
 
+        @Override
+        public void setColumnComment(Session session, TableHandle tableHandle, ColumnHandle column, Optional<String> comment)
+        {
+            ConnectorTableMetadata tableMetadata = getTableMetadata(tableHandle);
+            TestingColumnHandle columnHandle = (TestingColumnHandle) column;
+            ConnectorTableMetadata newTableMetadata = new ConnectorTableMetadata(
+                    tableMetadata.getTable(),
+                    tableMetadata.getColumns().stream()
+                            .map(tableColumn -> Objects.equals(tableColumn.getName(), columnHandle.getName()) ? withComment(tableColumn, comment) : tableColumn)
```
Comment (ebyhr) [inline core/trino-main/src/test/java/io/trino/execution/BaseDataDefinitionTaskTest.java:438]: Why not `tableColumn.getName().equals(columnHandle.getName())`?

File: core/trino-main/src/test/java/io/trino/execution/BaseDataDefinitionTaskTest.java [hunk 3]
```diff
@@ -433,5 +455,18 @@ public ResolvedFunction getCoercion(Session session, OperatorType operatorType,
         {
             return delegate.getCoercion(session, operatorType, fromType, toType);
         }
+
+        private static ColumnMetadata withComment(ColumnMetadata tableColumn, Optional<String> comment)
+        {
+            return ColumnMetadata.builder()
```
Comment (ebyhr) [inline core/trino-main/src/test/java/io/trino/execution/BaseDataDefinitionTaskTest.java:461]: How about using `ColumnMetadata.builderFrom`? 
```java
            return ColumnMetadata.builderFrom(tableColumn)
                    .setComment(comment)
                    .build();
```

File: core/trino-main/src/test/java/io/trino/execution/TestCommentTask.java [hunk 4]
```diff
@@ -106,6 +107,22 @@ public void testCommentViewOnMaterializedView()
...
     }
 
+    @Test
+    public void testCommentTableColumn()
+    {
+        QualifiedObjectName tableName = qualifiedObjectName("existing_table");
+        QualifiedName columnName = qualifiedColumnName("existing_table", "test");
+        metadata.createTable(testSession, TEST_CATALOG_NAME, someTable(tableName), false);
+        assertThat(metadata.getTableMetadata(testSession, metadata.getTableHandle(testSession, tableName).get()).getMetadata().getComment())
+                .isEmpty();
```
Comment (ebyhr) [inline core/trino-main/src/test/java/io/trino/execution/TestCommentTask.java:117]: We can remove this assertion as the test purpose is verifying column comment, not table comment.

**Review by ebyhr — APPROVED** [review]

File: core/trino-main/src/test/java/io/trino/execution/BaseDataDefinitionTaskTest.java (hunk 3 above)
Comment (findinpath) [reply core/trino-main/src/test/java/io/trino/execution/BaseDataDefinitionTaskTest.java:461]: I missed this utility method. Thanks for pointing it out.

**Comment by ebyhr** [comment]
Merged, thanks!

**Pull request merged**